### ISSUE TITLE
Could you offer the pre-extracted features of msrvtt and tgifqa?

### ISSUE THREAD

**Issue by LemonQC** [doc-doc/HQGA#4]

**Comment by doc-doc** [comment]
Hi, the features are too large to share online, please use the links provided at the bottom of readme to extract features for them.

**Comment by llh11211hll** [comment]
Hi, so sorry to bother you, I wonder if it is convenient to share the extracted BERT feature on msrvtt (if those on tgifqa are too large),Could I get a download link?This is my email (2422008572@qq.com).Thanks a lot!!!

**Comment by doc-doc** [comment]
Hi, please download the fine-tuned BERT feature for MSRVTT-QA [here](https://drive.google.com/file/d/1gw10e2AziRel9TvzF_GVzKJYjAdhaLQ1/view?usp=sharing). Note that they are strictly aligned with  the privided QA annotations under ./datasets/msrvtt.

**Comment by llh11211hll** [comment]
> Hi, please download the fine-tuned BERT feature for MSRVTT-QA [here](https://drive.google.com/file/d/1gw10e2AziRel9TvzF_GVzKJYjAdhaLQ1/view?usp=sharing). Note that they are strictly aligned with the privided QA annotations under ./datasets/msrvtt.

Thank you very much for your generous help!

**Comment by LemonQC** [comment]
I wonder how much memory does the next-qa need? I tried on 64G with 3090, the memory is out. Thanks.

**Comment by LemonQC** [comment]
> 

Well, Thanks any way. If you could just offer the region feature, I think it's ok

**Comment by doc-doc** [comment]
Hi, the region feature is provided in the readme.  It needs about 24G GPU memory with 8 clips for each video.

**Comment by LemonQC** [comment]
There exists the region feature for Next-qa and MSVD-QA, I wonder if you can share the remaining two. My GPU is limited and hard to extract the feature. Thanks.

**Comment by LemonQC** [comment]
Moreover, in your sparse sampling, there should be 0.25*16=4 frames in region feature, while there exists 16.
![image](https://user-images.githubusercontent.com/30914380/167850608-7001e48d-7507-4aae-84e5-0c886f20b7bf.png)

**Comment by doc-doc** [comment]
Hi, the tensor shape means that there are 570 videos with each 16 clips, and each clip has 4 (=0.25*16) frames with each frame 20 region proposals.
---------------
I do not have enough online space to share the features of the other two as they are too large.

**Comment by doc-doc** [comment]
1. Hi, the tensor shape means that there are 570 videos with each 16 clips, and each clip has 4 (=0.25*16) frames with each frame 20 region proposals.
2. I do not have enough online space to share the features of the other two datasets as they are too large. By the way, feature extract won't take so much memory, you may need to check you code or some hyper-parameters (eg. , batch size ...)

**Comment by doc-doc** [comment]
Hi, the tensor shape means that there are 570 videos with each 16 clips, and each clip has 4 (=0.25*16) frames with each frame 20 region proposals.
I do not have enough online space to share the features of the other two datasets as they are too large. By the way, feature extraction won't take so much memory, you may need to check your code or some hyper-parameters (eg. , batch size ...)

**Comment by LemonQC** [comment]
Well，many Thanks.  Could you provide the code for sampling. I think the sparse sampling may be effected the random seed. For example, in Next-QA, K=16, p=6, frame=4,does you sample 4 from the 6 in each clip(in total 16).




| |
Shandong University
|
|
***@***.***
|
On 5/11/2022 21:20，Junbin ***@***.***> wrote：

Hi, the tensor shape means that there are 570 videos with each 16 clips, and each clip has 4 (=0.25*16) frames with each frame 20 region proposals.
I do not have enough online space to share the features of the other two datasets as they are too large. By the way, feature extraction won't take so much memory, you may need to check your code or some hyper-parameters (eg. , batch size ...)

**Comment by doc-doc** [comment]
Hi, please refer to HCRN repo for feature sampling, we use the exact way as in HCRN. For video extraction, you can use ffmpeg or skvideo and extract them at a frame rate of 6 fps.

**Comment by LemonQC** [comment]
OK， many thanks any way. I will try it.


| |
Shandong University
|
|
***@***.***
|
On 5/12/2022 10:49，Junbin ***@***.***> wrote：

Hi, please refer to HCRN repo for feature sampling, we use the exact way as in HCRN. For video extraction, you can use ffmpeg or skvideo and extract them at a frame rate of 6 fps.

**Comment by LemonQC** [comment]
> Hi, please refer to HCRN repo for feature sampling, we use the exact way as in HCRN. For video extraction, you can use ffmpeg or skvideo and extract them at a frame rate of 6 fps.

I am very sorry to bother you again. While I found in HRCN, the num_frames_per_clip is only set to 16 frames. In the sparse sampling of HQGA,  the length is γL, well. I suppose you extract all the frame features and randomly select the γL features during each sampling process. Or you change another pretrained model to obtain the region feature?

**Comment by doc-doc** [comment]
Yes, once you have the tensor of 16 frames, you can obtain the final feature by sampling with app_feat[::4]. Region feature is from BUTD (see my link in readme..). You need to keep the same sampling strategy.

**Comment by LemonQC** [comment]
Well. Thanks again..         


| |
Shandong University
|
|
***@***.***
|
On 5/12/2022 15:12，Junbin ***@***.***> wrote：

Yes, once you have the tensor of 16 frames, you can obtain the final feature by sampling with app_feat[::4]. Region feature is from BUTD (see my link in readme..). You need to keep the same sampling strategy.
